### ISSUE TITLE
`conventional-gitlab-releaser` for GitLab

### ISSUE THREAD

**Issue by hutson** [conventional-changelog-archived-repos/meta#5]
I recently wrote `conventional-gitlab-releaser` to provide me the same functionality as `conventional-github-releaser`, but for the GitLab SCM platform.

You can find the repo here: https://gitlab.com/hutson/conventional-gitlab-releaser

This required pushing a minor change to `node-gitlab` to support adding new tags (and therefore new releases): https://github.com/node-gitlab/node-gitlab/pull/127

My long-term goal is to support a `semantic-release` like tool for GitLab and GitLab CI.

I wanted to reach out to y'all and see if you'd be interested in `conventional-gitlab-releaser`, or with supporting the GitLab platform.

I haven't published `conventional-gitlab-releaser` to npm yet, preferring to wait until I got your thought on the matter.

**Comment by stevemao** [comment]
Wonderful, add the tool to "Related" section please 😄 
We should probably create `conventional-releaser` to wrap two together and intelligently use one or the other based on the project.

**Comment by hutson** [comment]
I'll submit a pull request to add `conventional-gitlab-releaser` to the _Related_ section of `conventional-github-releaser`.

So, actually, you bring up a point that's been concerning me a little. About 90% of the code for `conventional-gitlab-releaser` is identical to `conventional-github-releaser`. I only changed how the client is constructed, and I replaced the line of code that called into the github module to create the release (with a call to gitlab.projects.repository.addTag). 

Because of the extensive redundancy in code between the two projects, it's quite likely that `conventional-gitlab-releaser` will fall out of date on occasion.

What if `conventional-releaser` was to take an approach similar to `conventional-changelog`? We could have projects like `conventional-gitlab-releaser` export an object that implements three methods:
- createClient
- authenticateClient
- publishRelease

Then `conventional-releaser` can depend on those packages and use them as it deems appropriate.
